pkg/image/importer/credentials.go

```diff
@@ -69,31 +67,50 @@ func (s *keyringCredentialStore) Basic(url *url.URL) (string, string) {
 	return basicCredentialsFromKeyring(s.DockerKeyring, url)
 }
 
-func NewCredentialsForSecrets(secrets []kapi.Secret) auth.CredentialStore {
-	return &secretCredentialStore{secrets: secrets}
+func NewCredentialsForSecrets(secrets []kapi.Secret) *SecretCredentialStore {
+	return &SecretCredentialStore{secrets: secrets}
 }
 
-type secretCredentialStore struct {
-	secrets []kapi.Secret
+func NewLazyCredentialsForSecrets(secretsFn func() ([]kapi.Secret, error)) *SecretCredentialStore {
+	return &SecretCredentialStore{secretsFn: secretsFn}
```

Comment:
I don't see that being any clearer - it's more wrappers and more things to go wrong.